### DISCUSSION TITLE
如何完全复制一个玩家数据到另一个玩家身上

### DISCUSSION THREAD

**Discussion by Leovany** [kbengine/kbengine#1377] [Q&A]
1） 场景
&nbsp;&nbsp;&nbsp;&nbsp;游戏上线之后，有的玩家数据出现了问题，这时候需要通过登录问题玩家的账号，进行排查问题
2）目前做法
&nbsp;&nbsp;&nbsp;&nbsp;     2.1 直接登录玩家账号排查，
&nbsp;&nbsp;&nbsp;&nbsp;          弊：期间玩家不能登录，也会破坏问题现场
&nbsp;&nbsp;&nbsp;&nbsp;     2.2 导出线上服务器的数据，到自己的测试服上，然后在登录排查
&nbsp;&nbsp;&nbsp;&nbsp;         弊：需拷贝全部数据，并且操作麻烦，容易误操作（比如手误，清除数据库....）
3）我目前想法
&nbsp;&nbsp;&nbsp;&nbsp;     通过后台，传入两个玩家ID, 将玩家1数据完全拷贝到玩家2身上，然后登录玩家2账号排查
&nbsp;&nbsp;&nbsp;&nbsp;具体做法：
    3.1 深拷贝 （）
&nbsp;&nbsp;&nbsp;&nbsp;        player_2 = copy.deepcopy(player_1) 
   3.2 每个模块单独拷贝
&nbsp;&nbsp;&nbsp;&nbsp;      player_2.itemData = player_1.itemData 
&nbsp;&nbsp;&nbsp;&nbsp;      player_2.moneyData = player_1.moneyData 
&nbsp;&nbsp;&nbsp;&nbsp;      等等.....

 3.1 深拷贝方法是最简单的，但我测试发现不成功，3.2 模块单独拷贝，功能太多，一个一个写麻烦了，而且容易遗漏
所以想请教一下大家，是否有更好的方式呢？
不胜感激！！！

**Comment by worldg** [comment]
可以考虑base实体的onWriteToDB接口调用时获得存档数据（可以调用writeToDB触发）， 然后在你自己账号实体下线调用onWriteToDB时赋值过去，再上线。

**Reply by Leovany** [reply]
有点没看懂，onWriteToDB 时， 两个账号该怎么关联上？
